templates/configmap-env-pre.yaml

```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "mastodon.fullname" . }}-env
```

Comment:
why do we need a second configmap with the same name and data?

Reply:
Because Helm doesn't create the other configmap until after the pre-install hooks, and deletes this one after running pre-install hooks. It's a catch-22: for the mastodon-env configmap to be available when db-migrate job runs, it has to be created as a hook with a lower weight, but since it's created as a hook it gets cleaned up after db-migrate is done, so we also have to keep the non-hook version of the same configmap. See also commit message in a7496545f6defce9cd395e934c0ba6c67aacc5de.

Reply:
oh i didn't realize that! thanks for the clarification.

Reply:
For some additional clarification: this is only about the first time the chart gets installed after that it will be there.